### PR TITLE
added healthcheck for reviewer service

### PR DESCRIPTION
**Pull request by Fidel1985**

### CODE REVIEW

File: DataProviders/src/main/java/com/softserveinc/reviewer/health/ElasticSearchHealthCheck.java [hunk 1]
```diff
@@ -1,33 +1,12 @@
...
-        Invocation.Builder invocationBuilder = webTarget.request(MediaType.APPLICATION_JSON);
-        Response response = invocationBuilder.get();
-        ElasticSearchResponse elasticSearchResponse = response.readEntity(ElasticSearchResponse.class);
-
-        if (elasticSearchResponse != null && elasticSearchResponse.getHits() != null) {
-            return Result.healthy();
-        }
-        return Result.unhealthy("ElasticSearch API Failed.");
+        return Result.healthy("ElasticSearch Service is healthy!");
+        //return Result.unhealthy("something went wrong");
```
Comment (svoronin) [inline DataProviders/src/main/java/com/softserveinc/reviewer/health/ElasticSearchHealthCheck.java]: Please remove that comment

**Review by svoronin — CHANGES_REQUESTED** [review]
Overall I think that we don't need entities for healthcheck responses. We should be fine with checking HTTP code responses. If that is 200, we consider related service as healthy. Otherwise we consider it down or unavailable (if responses take too long). Also I think we can have only one common implementation for such HealthCheck and instantiate it for every downstream service with proper healthCheckURL.

**Comment by Fidel1985** [comment]
If we would check only response codes, in case when one of the services is unhealthy, the hole response would have status code 500 (Server Error). So I created that entities to localize unheathy service.

File: Reviewer/configuration.yaml [hunk 2]
```diff
@@ -6,11 +6,14 @@ server:
   - type: http
     port: 8083
 
+syndicationAdminBaseUrl: http://localhost:8085
```
Comment (svoronin) [inline Reviewer/configuration.yaml]: I'd better call it ***HealthcheckUrl

**Comment by svoronin** [comment]
I think it is legit to return 500 in case any of the downstream services is not working properly

File: Reviewer/src/main/java/com/softserveinc/reviewer/health/DataProvidersHealthCheck.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    private final String healthCheckUrl;
+
+    @Inject
+    public DataProvidersHealthCheck(String healthCheckUrl) {
+        this.healthCheckUrl = healthCheckUrl;
+    }
+
+    @Override
+    protected Result check() throws Exception {
+        URL url = new URL(healthCheckUrl + "/healthcheck");
```
Comment (svoronin) [inline Reviewer/src/main/java/com/softserveinc/reviewer/health/DataProvidersHealthCheck.java]: `/health` should be part of field value in configuration

File: Reviewer/src/main/java/com/softserveinc/reviewer/ReviewerApplication.java [hunk 4]
```diff
@@ -22,7 +22,9 @@ public String getName() {
     @Override
     public void run(final ReviewerConfiguration configuration, final Environment environment) {
         Injector injector = Guice.createInjector(new GuiceModule(configuration));
-        environment.healthChecks().register("ReviewerHealthCheck", injector.getInstance(ReviewerHealthCheck.class));
+        environment.healthChecks().register("SyndicationHealthCheck", new DataProvidersHealthCheck(configuration.getSyndicationHealthCheckUrl()));
+        environment.healthChecks().register("OracleHealthCheck", new DataProvidersHealthCheck(configuration.getOracleHealthCheckUrl()));
+        environment.healthChecks().register("ElasticSearchHealthCheck", new DataProvidersHealthCheck(configuration.getElasticSearchHealthCheckUrl()));
```
Comment (svoronin) [inline Reviewer/src/main/java/com/softserveinc/reviewer/ReviewerApplication.java]: Please change it to be switchboard, oracle and elasticsearch respectively. No need to call it health check explicitly.

**Review by svoronin — APPROVED** [review]

**Pull request merged**